src/index.spec.js

```diff
@@ -197,6 +199,12 @@ describe('provideState', () => {
       expect(sum).toHaveBeenCalledTimes(3)
     })
 
+    it('throws when a computed calls its self', () => {
+      expect(() => {
+        return getInjectedState().circularComputed
+      }).toThrowError(new Error('computed: circularComputed, cannot access computed from itself'))
```

Comment:
How will it work with #24?

Reply:
maybe we can type errors and do something like this
```js
try {
                    previousValue = c(stateProxy, propsProxy)
                  } catch (error) {
                    if (error instanceof ComputedError) {
                      throw error
                    }
                    noop()
                  }
```

Reply:
I'm not very fond of this but I don't see better solutions.